### PR TITLE
Added hiddenOnStartup flag to module config

### PR DESCRIPTION
**Pull request by MystaraTheGreat**
This adds a 'hiddenOnStartup' flag to module configurations.  The flag (which is false as default) causes the module to be hidden (via hide() ) after it is started up.  This is useful for situations where another mechanism exists to cause a module to appear (e.g. MMM-Modulebar and others).

This addresses the feature request https://github.com/MichMich/MagicMirror/issues/2470

### CODE REVIEW

**Comment by MystaraTheGreat** [comment]
I have no idea what "Travis CI - Pull Request Expected — Waiting for status to be reported" means.  I'm assuming it will magically sort itself out in due course :)

**Comment by rejas** [comment]
Please base your PR against the develop branch

**Comment by MystaraTheGreat** [comment]
> Please base your PR against the develop branch

I think that's now done.

File: js/loader.js [hunk 1]
```diff
@@ -54,6 +54,16 @@ var Loader = (function () {
 
 		// Notify core of loaded modules.
 		MM.modulesStarted(moduleObjects);
+
+		// Starting modules also hides any modules that have requested to be initially hidden
+		for (var n in moduleObjects) {
```
Comment (rejas) [inline js/loader.js:59]: var is outdated, please use let (or const if its a cosntant)

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ _This release is scheduled to be released on 2021-04-01._
 - `module.show` has now the option for a callback on error.
 - Added locale to sample config file
 - Added support for self-signed certificates for the default calendar module (#466)
+- Added hiddenOnStartup flag to module config (#2475)
```
Comment (rejas) [inline CHANGELOG.md:25]: You should also consider adding a PR for the docs: https://docs.magicmirror.builders/modules/configuration.html

**Review by rejas — CHANGES_REQUESTED** [review]
Besides the var thing it looks good to me.

File: js/loader.js (hunk 1 above)
Comment (MystaraTheGreat) [reply js/loader.js]: Okay, I think that's now done

File: js/loader.js [hunk 3]
```diff
@@ -54,6 +54,16 @@ var Loader = (function () {
 
 		// Notify core of loaded modules.
 		MM.modulesStarted(moduleObjects);
+
+		// Starting modules also hides any modules that have requested to be initially hidden
+		for (let n in moduleObjects) {
+			let thisModule = moduleObjects[n];
```
Comment (rejas) [inline js/loader.js:60]: i think you could write it like this:
for (let thisModule of moduleObjects) {

Comment (MystaraTheGreat) [reply js/loader.js]: I.....yes.
Dammit.

:)

**Review by rejas — APPROVED** [review]

**Comment by rejas** [comment]
Looks good. When its merged you can look into writing a PR for the docs and (if you dare ;-) a PR for testing the feature :-)

**Comment by MystaraTheGreat** [comment]
> Looks good. When its merged you can look into writing a PR for the docs and (if you dare ;-) a PR for testing the feature :-)

Already did a PR for the docs :)

PR for testing...uhhhhhh "maybe", haha

**Pull request merged**